### PR TITLE
Update unit tests to use `AEPTestUtils`

### LINKED ISSUE #41: Refactor functional and integration tests to use new test utils
### Task description

The existing functional and integration tests should be refactored to use the newly available common test utilities available in: https://github.com/adobe/aepsdk-testutils-ios

With the following items covered:
1. Any changes required to the common test utils should be opened as a PR to the test utils repo, optimally in a non-breaking change manner (although before the 1.0.0 release of the test utils, larger changes should be ok)
2. Manual dictionary/array property assertions should be refactored to use the JSON comparison tool
3. Test utilities that exist in the test utils repo should be used over equivalent test util classes defined locally

### Additional implementation details or code snippet(s)

_No response_

### PR DESCRIPTION
**Pull request by timkimadobe**
## Description

This PR is a continuation of the review process in #42 and refactors existing unit tests to use the `AEPTestUtils` and JSON comparison methods with the following changes:

**MediaPublicTrackerTests.swift**
1. All custom `isEqual` comparisons updated to use JSON comparison instead of custom dictionary casting and equality

**MediaXDMEventTests**
* testToXDMData - updated to use JSON comparison with strict equality
* For property `xdm.mediaCollection.sessionDetails.length` Is standard Int conversion fine (via AnyCodable decoder) instead of specific conversion to Int64?

**MediaXDMEventHelperTests**
1. Replace AssertUtils.compareSizeAndKeys with JSON comparison assertEqual

**XDMAdvertisingDetailsTests
XDMAdvertisingPodDetailsTests
XDMErrorDetailsTests
XDMChapterDetailsTests
XDMSessionDetailsTests
XDMMediaCollectionTests**
* `testEncode` does not have strict collection count equality validation so assertExactMatch was used - is exact equality desired instead?
    * Please note that as long as expected is valid, `actual` will be validated accordingly so guard/unwrap of `actual` is not strictly needed, unless there is a specific step that should be validated in depth

## Questions for reviewers
1. For dictionary value comparisons using the JSON comparison methods, `Double` values dont use the delta comparison logic present in `MediaPublicTrackerTests` `isEqual` method. Is this an acceptable change?:

from Sources -> Double+Media.swift:
```swift
// Adds functionality to compare the equality of two doubles to within the specified delta.
extension Double {
    func isAlmostEqual(_ doubleToCompare: Double, accuracy: Double = 0.000001) -> Bool {
        return fabs(self - doubleToCompare) < accuracy
    }
}
```
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Tests/UnitTests/XDMAdvertisingDetailsTests.swift [hunk 1]
```diff
@@ -29,22 +30,29 @@ class XDMAdvertisingDetailsTests: XCTestCase {
...
+          "length": 10,
+          "name": "id",
+          "placementID": "test_placementID",
+          "playerName": "test_playerName",
+          "podPosition": 1,
+          "siteID": "test_siteID"
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedAdDetails)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMAdvertisingDetailsTests.swift]: I'd suggest using `assertEqual` here to make the test more strict. If the class under test is updated in the future, this test will fail causing the developer to update the tests with the changes.

File: Tests/UnitTests/XDMAdvertisingPodDetailsTests.swift [hunk 2]
```diff
@@ -11,23 +11,31 @@
...
+        // Verify
+        let expected = """
+        {
+          "friendlyName": "name",
+          "index": 1,
+          "offset": 2
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedAdBreakDetails)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMAdvertisingPodDetailsTests.swift]: Use `assertEqual` here.

File: Tests/UnitTests/XDMErrorDetailsTests.swift [hunk 3]
```diff
@@ -11,22 +11,30 @@
...
-        XCTAssertEqual("test_errorSource", map["source"] as! String)
+        // Verify
+        let expected = """
+        {
+          "name": "test_errorID",
+          "source": "test_errorSource"
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedErrorDetails)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMErrorDetailsTests.swift]: Use `assertEqual` here.

File: Tests/UnitTests/XDMMediaCollectionTests.swift [hunk 4]
```diff
@@ -44,60 +45,75 @@ class XDMMediaCollectionTests: XCTestCase {
...
+            "segment": "test_segment",
+            "show": "test_show",
+            "showType": "test_showType",
+            "streamFormat": "test_streamFormat",
+            "streamType": "video"
+          }
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedMediaCollection)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMMediaCollectionTests.swift]: Use `assertEqual`

File: Tests/UnitTests/XDMMediaCollectionTests.swift [hunk 5]
```diff
@@ -44,60 +45,75 @@ class XDMMediaCollectionTests: XCTestCase {
...
+        {
+          "advertisingPodDetails": {
+            "friendlyName": "name",
+            "index": 1,
+            "offset": 2
+          }
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedMediaCollection)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMMediaCollectionTests.swift]: Use `assertEqual`

File: Tests/UnitTests/XDMMediaCollectionTests.swift [hunk 6]
```diff
@@ -112,42 +128,60 @@ class XDMMediaCollectionTests: XCTestCase {
...
+            "name": "id",
+            "placementID": "test_placementID",
+            "playerName": "test_playerName",
+            "podPosition": 1,
+            "siteID": "test_siteID"
+          }
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedMediaCollection)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMMediaCollectionTests.swift]: Use `assertEqual`

File: Tests/UnitTests/XDMMediaCollectionTests.swift [hunk 7]
```diff
@@ -112,42 +128,60 @@ class XDMMediaCollectionTests: XCTestCase {
...
+          "chapterDetails": {
+            "friendlyName": "name",
+            "index": 1,
+            "length": 10,
+            "offset": 2
+          }
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedMediaCollection)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMMediaCollectionTests.swift]: Use `assertEqual`

File: Tests/UnitTests/XDMMediaCollectionTests.swift [hunk 8]
```diff
@@ -164,14 +198,27 @@ class XDMMediaCollectionTests: XCTestCase {
...
+              "name": "test_mute"
+            },
+            {
+              "name": "test_fullscreen"
+            }
+          ]
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedMediaCollection)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMMediaCollectionTests.swift]: Use `assertEqual`

File: Tests/UnitTests/XDMMediaCollectionTests.swift [hunk 9]
```diff
@@ -181,13 +228,26 @@ class XDMMediaCollectionTests: XCTestCase {
...
```
Comment (kevinlind) [inline Tests/UnitTests/XDMMediaCollectionTests.swift]: Use `assertEqual`

File: Tests/UnitTests/XDMSessionDetailsTests.swift [hunk 10]
```diff
@@ -43,45 +44,50 @@ class XDMSessionDetailsTests: XCTestCase {
...
+          "season": "1",
+          "segment": "test_segment",
+          "show": "test_show",
+          "showType": "test_showType",
+          "streamFormat": "test_streamFormat",
+          "streamType": "video"
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedSessionDetails)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMSessionDetailsTests.swift]: Use `assertEqual`

File: Tests/UnitTests/XDMSessionDetailsTests.swift [hunk 11]
```diff
@@ -94,28 +100,33 @@ class XDMSessionDetailsTests: XCTestCase {
...
+          "length": 30,
+          "name": "id",
+          "playerName": "test_playerName",
+          "publisher": "test_publisher",
+          "station": "test_station",
+          "streamType": "audio"
+        }
+        """
+
+        assertExactMatch(expected: expected, actual: decodedSessionDetails)
```
Comment (kevinlind) [inline Tests/UnitTests/XDMSessionDetailsTests.swift]: Use `assertEqual`

**Comment by kevinlind** [comment]
@timkimadobe , regarding your question on using `isEqual` instead of the custom Double.isAlmostEqual in `MediaPublicTrackerTests`, are you seeing any test failures using `isEqual`?
It appears these tests pass in the exact values expected, so there shouldn't be any difference between the actual and expected values. The `isAlmostEqual` logic might be overkill in this case.

File: Tests/UnitTests/XDMAdvertisingDetailsTests.swift (hunk 1 above)
Comment (timkimadobe) [reply Tests/UnitTests/XDMAdvertisingDetailsTests.swift]: Updated to use `assertEqual` across the board!

**Review by timkimadobe — COMMENTED** [review]
Thanks for the review @kevinlind! Updated to use `assertEqual` across the board, with the `XDMMediaCollectionTests` -> `testEncode_sessionStart()` requiring two additional properties added to the expected JSON as they were not present in the original test case

**Review by kevinlind — APPROVED** [review]

**Pull request merged**